apt/method.go

```diff
@@ -0,0 +1,225 @@
+package apt
+
+import (
+	"bufio"
+	"context"
+	"crypto/md5"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+)
+
+func NewAptMethod(output io.Writer, input *bufio.Reader) *AptMethod {
+	return &AptMethod{
+		config: &AptMethodConfig{},
```

Comment:
should config be an arg as well for future extensibility?

Reply:
this config is not provided by user but is where we store the result of parsing a `601 Configure` message